libs/agno/agno/observability/langfuse.py

```diff
@@ -0,0 +1,123 @@
+from dataclasses import dataclass
+from datetime import datetime
+from os import getenv
+from typing import Any, Dict, Optional
+
+try:
+    from langfuse import Langfuse
+except ImportError:
+    raise ImportError("`langfuse` not installed. Please install using `pip install langfuse`")
+
+
+@dataclass
+class LangfuseObservability:
+    public_key: Optional[str] = None
+    secret_key: Optional[str] = None
+    host: Optional[str] = None
+    release: Optional[str] = None
+    debug: Optional[bool] = False
+    threads: Optional[int] = None
+    max_retries: Optional[int] = None
+    timeout: Optional[int] = None
+    sample_rate: Optional[float] = None
+
+    request_params: Optional[Dict[str, Any]] = None
+    client: Optional[Langfuse] = None
+
+    _current_trace_id: Optional[str] = None
+    _current_span_id: Optional[str] = None
+    _current_generation_id: Optional[str] = None
+
+    def get_client(self):
+        if self.client is not None:
+            return self.client
+
+        self.public_key = self.public_key or getenv("LANGFUSE_PUBLIC_KEY")
+        self.secret_key = self.secret_key or getenv("LANGFUSE_SECRET_KEY")
+
+        if not self.public_key or not self.secret_key:
+            raise ValueError("LANGFUSE_PUBLIC_KEY and LANGFUSE_SECRET_KEY must be set")
+
+        self.host = self.host or getenv("LANGFUSE_HOST")
+        self.release = self.release or getenv("LANGFUSE_RELEASE")
+        self.debug = self.debug or getenv("LANGFUSE_DEBUG")
+        self.threads = self.threads or getenv("LANGFUSE_THREADS")
+        self.max_retries = self.max_retries or getenv("LANGFUSE_MAX_RETRIES")
+        self.timeout = self.timeout or getenv("LANGFUSE_TIMEOUT")
+        self.sample_rate = self.sample_rate or getenv("LANGFUSE_SAMPLE_RATE")
+
+        self.request_params = self.request_params or {}
+
+        langfuse = Langfuse(
+            public_key=self.public_key,
+            secret_key=self.secret_key,
+            host=self.host,
+            release=self.release,
+            debug=self.debug,
+            threads=self.threads,
+            max_retries=self.max_retries,
+            timeout=self.timeout,
+            sample_rate=self.sample_rate,
+        )
+        self.client = langfuse
+        return self.client
+
+    def trace(
+        self,
+        name: Optional[str] = None,
+        input_dict: Optional[Dict[str, Any]] = None,
+    ):
```

Comment:
Wondering if we should enforce the trace id and session_id to be our agent's run_id and session_id.
The idea of the sessions matching sounds especially useful. They also seem to recommend/push for this judging by the docstring in `Langfuse.trace()`:
```
id: The id of the trace can be set, defaults to a random id. Set it to link traces to external systems or when creating a distributed trace. Traces are upserted on id.
session_id: Used to group multiple traces into a session in Langfuse. Use your own session/thread identifier.
```

Reply:
Ah good spot! I'll add